src/main/java/org/sagebionetworks/web/client/GlobalApplicationStateImpl.java

```diff
@@ -570,19 +570,32 @@ public void onPreviewNativeEvent(NativePreviewEvent event) {
             );
             // SWC-5812: SRC has components that apply this special CSS class, and expect the app to look for these events and handle appropriately.
             // When the class name is undefined, targetElement.getClassName() still returns an object where isEmpty is false and is not equal to null.
+            boolean isSRCSignInClass = false;
             try {
-              if (
+              isSRCSignInClass =
                 targetElement.hasClassName("SRC-SIGN-IN-CLASS") &&
-                !ginInjector.getAuthenticationController().isLoggedIn()
-              ) {
-                getPlaceChanger().goTo(new LoginPlace(LoginPlace.LOGIN_TOKEN));
-              }
+                !ginInjector.getAuthenticationController().isLoggedIn();
             } catch (Exception e) {
               // SWC-6243: log the error if it is not the known problem of finding the className
               if (!e.getMessage().contains("indexOf")) {
                 synapseJSNIUtils.consoleError(e.getMessage());
               }
             }
+            try {
+              if (isSRCSignInClass) {
+                getPlaceChanger().goTo(new LoginPlace(LoginPlace.LOGIN_TOKEN));
+              } else {
+                if (targetElement.hasAttribute("href")) {
+                  String href = targetElement.getAttribute("href");
+                  boolean handled = handleRelativePathClick(href);
+                  if (handled) {
+                    event.cancel();
+                  }
+                }
+              }
+            } catch (Exception e) {
+              synapseJSNIUtils.consoleError(e.getMessage());
+            }
```

Comment:
Does this work with the tabs in the `DataAccessManagement` place (ACT/AR reviewer dashboard), where the subpath is controlled by React Router?

Reply:
Yes, verified that this works as expected, thanks for asking.
There is an uncaught exception whenever I switch tabs in this place, but that's true on production today as well (unexpected null).
<img width="977" alt="Screen Shot 2024-06-12 at 2 33 34 PM" src="https://github.com/Sage-Bionetworks/SynapseWebClient/assets/1864447/762f7621-4f55-4369-a78b-425fe53741ed">